# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/html/nsGenericHTMLElement.h
# Commit: acdd013202e1
# Full Hash: acdd013202e1e688986e7e6c747b7c7fee38474e
# Author: Boris Chiou <boris.chiou@gmail.com>
# Date: 2022-08-23 02:46:57
# Regressor Bug: 1694741
# File Overlap Count: 1
# Description:
#   Bug 1694741 - Part 6: Map width/height attributes to the style of img elements. r=emilio
#   
#   This patch will use the width/height attributes from <source> to override
#   width/height/aspect-ratio CSS property values of <img> elements.
#   
# ==============================================================================

diff -r 2b4d9516b882 -r acdd013202e1 dom/html/nsGenericHTMLElement.h
--- a/dom/html/nsGenericHTMLElement.h	Mon Aug 22 20:18:39 2022 +0000
+++ b/dom/html/nsGenericHTMLElement.h	Mon Aug 22 20:18:39 2022 +0000
@@ -532,6 +532,16 @@
                                          mozilla::MappedDeclarations&,
                                          MapAspectRatio = MapAspectRatio::No);
   /**
+   * Helper to map the iamge source attributes into a style stuct.
+   * Note:
+   * This will override the declaration created by the presentation attributes
+   * of HTMLImageElement (i.e. mapped by MapImageSizeAttributeInto).
+   * https://html.spec.whatwg.org/multipage/embedded-content.html#the-source-element
+   */
+  static void MapPictureSourceSizeAttributesInto(const nsMappedAttributes*,
+                                                 mozilla::MappedDeclarations&);
+
+  /**
    * Helper to map the width and height attributes into the aspect-ratio
    * property.
    *